substantial/workflows/context.py

```diff
@@ -77,23 +91,24 @@ async def save(
 
         if saved is None:
             val = await evaluator.exec(self, save_id, None)
-            print(f"Computed id#{save_id}, {val}")
-            return val
+            parsed_val = parse(json.dumps(val), expected_type)
```

Comment:
Why this extra-step? Isn't the value returned as is on the first replay or..?

Reply:
my bad!